R/get_cdc_childvaxview_data.R

```diff
@@ -0,0 +1,83 @@
+#' Pull CDC NCIRD childvaxview immunization coverage data
+#' @description
+#' Pull coverage data from API and filter by desired geographic level
+#' @param limit `int` Number of rows to download, defaults to max allowed (1000).
+#' @param geo_level `str` Geographic categories of coverage data.
+#' Choose from: 'national', 'regional', 'state', or 'substate'.
+#'
+#' @return `tibble` Dataframe of vaccine coverage estimates for all VPDs.
+#' @export
+#'
+#' @examples
+#' cdc_data <- get_cdc_childvaxview_data(geo_level="substate")
+#'
+get_cdc_childvaxview_data <- function(limit = 1000, geo_level=NULL) {
+
+  # # Load necessary libraries
+  # library(httr)
+  # library(jsonlite)
+
+  ## Get data from API (>1000 rows) ####
+  base_url <- "https://data.cdc.gov/resource/fhky-rtsk.json" # NCIRD API endpoint
+
+  # Initialize variables for pagination
+  offset <- 0
+  all_data <- list() # Create a list to store data
+
+  repeat {
+    url <- sprintf("%s?$limit=%d&$offset=%d", base_url, limit, offset)
+    response <- httr::GET(url)
+
+    # Check if request was successful (status code 200)
+    if (httr::status_code(response) == 200) {
+      data_chunk <- jsonlite::fromJSON(httr::content(response, "text")) # Parse JSON content
+
+      if (length(data_chunk) == 0) {
+        break # Exit loop if no more data is returned
+      }
+
+      all_data[[length(all_data) + 1]] <- data_chunk # Append new chunk of data
```

Comment:
could use `list.append()` instead of numbering the lists. You could also just `bind_row()` into a tibble and avoid putting anything into lists.